src/ngraph/op/softmax.hpp

```diff
@@ -22,36 +22,79 @@ namespace ngraph
+                /// \brief Constructs a softmax operation.
+                ///
+                /// \param arg Node that produces the first input tensor.<br>
+                /// `[d0, ...]`
+                /// \param axes The axis positions (0-based) on which to calculate the softmax.
+                ///
+                /// Output `[d0, ...]`
+                ///
+                Softmax(const Output<Node>& arg, const AxisSet& axes);
+
+                virtual std::shared_ptr<Node>
+                    copy_with_new_args(const NodeVector& new_args) const override;
+
+                const AxisSet& get_axes() const { return m_axes; }
+                void set_axes(const AxisSet& axes) { m_axes = axes; }
+            protected:
+                virtual void generate_adjoints(autodiff::Adjoints& adjoints,
+                                               const NodeVector& deltas) override;
+
+            private:
+                AxisSet m_axes;
+            };
+        }
+
+        namespace v1
+        {
+            class Softmax : public util::UnaryElementwiseArithmetic
+            {
+            public:
+                NGRAPH_API
+                static const std::string type_name;
+                const std::string& description() const override { return type_name; }
+                Softmax()
+                    : m_axis(0)
+                {
+                }
+                /// \brief Constructs a softmax operation.
+                ///
+                /// \param arg Node that produces the first input tensor.<br>
+                /// `[d0, ...]`
+                /// \param axis The axis position (0-based) on which to calculate the softmax.
+                ///
+                /// Output `[d0, ...]`
+                ///
+                Softmax(const Output<Node>& arg, const size_t axis);
+
+                size_t get_op_version() const override { return 1; }
+                virtual std::shared_ptr<Node>
+                    copy_with_new_args(const NodeVector& new_args) const override;
 
-            virtual std::shared_ptr<Node>
-                copy_with_new_args(const NodeVector& new_args) const override;
+                size_t get_axis() const { return m_axis; }
+                void set_axis(const size_t axis) { m_axis = axis; }
+            protected:
+                virtual void generate_adjoints(autodiff::Adjoints& adjoints,
+                                               const NodeVector& deltas) override;
 
-            const AxisSet& get_axes() const { return m_axes; }
-            void set_axes(const AxisSet& axes) { m_axes = axes; }
-        protected:
-            virtual void generate_adjoints(autodiff::Adjoints& adjoints,
-                                           const NodeVector& deltas) override;
+            private:
+                size_t m_axis;
+            };
+        }
 
-        private:
-            AxisSet m_axes;
-        };
+        // default opset version
+        using v0::Softmax;
```

Comment:
@aprocter Can we make `op::v0::Softmax` visible in namespace `op`?

Reply:
From technical point of view no, but when we will have more `opset` supported by backends it can be probably confused to decide which version should be visible in `op` namespace